_projects/go_bears_on_bsd.md

```diff
@@ -0,0 +1,62 @@
+name: "Linter Dependency Management from a Bare-Metal System"
+requirements:
+  - "At least one patch to the coala core should be accepted and merged."
+  - "The proposal needs to show familiarity with coala's dependency management implementation."
+status: in_progress
+tags:
+  - Automation
+  - Installation
+  - Dependency
+---
+
+`coala-bears` are of two types, Native Bears which implement all the business logic themselves and Linter Bears which wrap existing linters and checkers. Dependencies of bears are mostly language runtimes, language-specific packages (e.g. python:nltk) or command line tools (e.g. git). Both Native Bears and Linter Bears have dependencies which need to be installed before they can be executed.
+
+`dependency_management` aims to provide Python APIs for dependency management on various languages and different systems. It is used extensively by coala, however, it lacks most of the key implementation and documentation in its current state.
+
+coala now has [7 bears](https://github.com/coala/coala-bears/blob/master/bears/go/) for the Go language. They need specific versions of Go and some externally supplied packages, and currently Go linter bears just assume they are all in place and work. Go language dependencies are currently managed unversioned ("If you're using an externally supplied package and worry that it might change in unexpected ways, the simplest solution is to copy it to your local repository. This is the approach Google takes internally", see [Go Official FAQ](https://golang.org/doc/faq#get_version)).
+
+The end goal is to be able to bootstrap and recursively install all dependencies on any systems without anything installed other than a base system, but the scope of a complete solution is too large. The student made a proposal with suitable scope, which is to automate the installation of Go linter bears onto bare-metal BSDs.
+
+Features will mainly be implemented to coala's `dependency_management`. It should be able to operate system and language-specific package managers to handle dependencies. It should also resolve package names and versions automatically using Repology on multiple operating systems.
+
+#### Milestones
+
+##### PREPARATION/BONDING
+
+* The student has identified most of the related problems.
+* The student has came up with a workflow of how dependencies would be managed across multiple operating systems and package managers.
+
+##### CODING PHASE 1
+
+* The student has identified and compiled the dependencies of the Go Linter into `coala-bears`.
+* The student has deployed a personal FreeBSD CI server and linked it to the working repo.
+* The student has implemented support for FreeBSD's default package manager [pkg](https://github.com/freebsd/pkg) and test mocks in `dependency_management`.
+* The student has started implementing bootstrap features for bare-metal systems.
+
+##### CODING PHASE 2
+
+* The student has ensured Repology metadata for the Go Linter is in place and correct.
```

Comment:
I think phase one should have at least one link to freebsd via repology completed, or at least investigated. So we learn about problems early, and any change in milestones occurs at end of phase 1

Then phase two finishes the process.